### PR TITLE
[15.0][MIG] hr_attendance_geolocation: Migration to 15.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 15.0

Please @pedrobaeza and @CarlosRoca13 can you review it?

@Tecnativa TT40358

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Tested in runboat, no geolocation permission is asked, nor any location logged on attendance check:

Selección_004

By the way, there's no visualization of the coordinates (I needed to put the fields on the list view). Maybe a couple of computed field putting them on human readable format and showing them on the list for the managers?

**Comment by nimarosa** [comment]
It will be great if we can use a reverse geocoding library with this module, to show the address, and even better if we can show a map.

Just a suggestion, but I think this will be really useful if we have that.

**Comment by victoralmau** [comment]
> 
> By the way, there's no visualization of the coordinates (I needed to put the fields on the list view). Maybe a couple of computed field putting them on human readable format and showing them on the list for the managers?

Currently the information was only displayed in the form view.
ejemplo

**Comment by victoralmau** [comment]
IMO, I think a good solution would be:

Form view:
- Add 2 embedded maps from google maps with corresponding coordinates.

Tree view:
- Add 2 fields (`geolocation_check_in` and `geolocation_check_out`) with the link to google maps with the corresponding coordinates.

What do you think about this?

**Comment by pedrobaeza** [comment]
All the Google map stuff is out of scope of this module, which aim is to store the information in raw format.

**Comment by nimarosa** [comment]
> IMO, I think a good solution would be:
> 
> 
> 
> Form view:
> 
> - Add 2 embedded maps from google maps with corresponding coordinates.
> 
> 
> 
> Tree view:
> 
> - Add 2 fields (`geolocation_check_in` and `geolocation_check_out`) with the link to google maps with the corresponding coordinates.
> 
> 
> 
> What do you think about this?

Looks like a good solution but like Pedro said, maybe we have to do it out of this migration PR, opening another one. I made the suggestion here to see what you guys think. 

We should do it in a new PR or create another module which implements this.

**Comment by victoralmau** [comment]
Added more readable fields to list and form view.

**Comment by pedrobaeza** [comment]
Fields should be only visible to managers, and `raw` is the opposite to human-readable concept. And please check that the core feature (registering the coordinates) is not working AFAIK.

**Comment by victoralmau** [comment]
I have changed the way the coordinates are displayed. 
The module does allow to save the coordinates in check in and check out attendances but I think we could re-factor the javascript almost completely to reduce the code and use context to set coordinates.

File: hr_attendance_geolocation/models/hr_attendance.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+class HrAttendance(models.Model):
+    _inherit = "hr.attendance"
+
+    check_in_latitude = fields.Float(digits="Location", readonly=True)
+    check_in_latitude_raw = fields.Char(
+        "Check-in Latitude", compute="_compute_check_in_latitude_raw"
+    )
+    check_in_longitude = fields.Float(digits="Location", readonly=True)
+    check_in_longitude_raw = fields.Char(
```
Comment (pedrobaeza) [inline hr_attendance_geolocation/models/hr_attendance.py]: This is the opposite of raw. Please call it `check_in_longitude_human` or `check_in_longitude_text` or `check_in_longitude_readable`.

**Review by pedrobaeza — COMMENTED** [review]
I'm trying on runboat and still don't get that any geolocation is logged. Firefox is not asking me to allow geolocation. Am I doing something wrong?

**Comment by victoralmau** [comment]
> I'm trying on runboat and still don't get that any geolocation is logged. Firefox is not asking me to allow geolocation. Am I doing something wrong?

Oh, I just found out that you can **only request geolocation** in the browser **if you use https**.

If you try it in localhost you can check how it works.

**Comment by pedrobaeza** [comment]
Please document it on the README.

**Comment by pedrobaeza** [comment]
@sbidoul putting `https` previously on the URL, it worked and get runboat HTTPS, but now it seems it's not working. Maybe this was due to the change for making it work fast? Is there any way to do it?

**Review by pedrobaeza — APPROVED** [review]

**Comment by nimarosa** [comment]
Hello, I tested in my test instance with https and it works. I tested like 5 times. I don't know if it work always but for me and in chrome browser is working.

**Review by nimarosa — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Thanks for the extra testing, Nicolás. A pity that it can't be tested on runboat, although I'm waiting the answer from @sbidoul to see if there's any possibility.

/ocabot merge nobump

**Comment by sbidoul** [comment]
> @sbidoul putting https previously on the URL, it worked and get runboat HTTPS, but now it seems it's not working. Maybe this was due to the change for making it work fast? Is there any way to do it?

Ah, that's likely a subtle behavior change between nginx and haproxy.
There is no certificate configured on the ingress, so maybe nginx used a default one.

**Comment by pedrobaeza** [comment]
> Ah, that's likely a subtle behavior change between nginx and haproxy. There is no certificate configured on the ingress, so maybe nginx used a default one.

And is there any possibility to put a default one somehow to have HTTPS?

**Comment by sbidoul** [comment]
I'll have look. Don't hold your breath, though as I'm fairly busy these days.

**Comment by pedrobaeza** [comment]
Hehe, OK, your work on this is always appreciated.

**Comment by nimarosa** [comment]
> @pedrobaeza The merge process could not be finalized, because command `git push origin 15.0-ocabot-merge-pr-80-by-pedrobaeza-bump-nobump:15.0` failed with output:
> ```
> To https://github.com/OCA/hr-attendance
>  ! [rejected]        15.0-ocabot-merge-pr-80-by-pedrobaeza-bump-nobump -> 15.0 (fetch first)
> error: failed to push some refs to 'https://***@github.com/OCA/hr-attendance'
> hint: Updates were rejected because the remote contains work that you do
> hint: not have locally. This is usually caused by another repository pushing
> hint: to the same ref. You may want to first integrate the remote changes
> hint: (e.g., 'git pull ...') before pushing again.
> hint: See the 'Note about fast-forwards' in 'git push --help' for details.
> 
> ```

@pedrobaeza I'm seeing a lot of this errors in different PR. The bot finally makes the merge but with an error. Is this something happening in instances that is causing it? Just curious.

**Comment by pedrobaeza** [comment]
I can't really say. Maybe it's due to any bot commit meanwhile. @sbidoul may have a hint.

**Comment by sbidoul** [comment]
That's new to me, and we should worry about it, it's really abnormal. No time to look right now, though.

**Comment by sbidoul** [comment]
You can now manually force an https URL in runboat, and if you accept the default certificate it should work.

**Comment by pedrobaeza** [comment]
Thank you very much for the quick reaction, @sbidoul. Tested and working, although there's a weird initial redirection and has to be typed 2 times, but that's something of Odoo itself I think.

@victoralmau now that I have been able of testing it, the human representation puts both on longitude and latitude the word `N/S` instead of `E/W` in the second one.

**Pull request merged**